shared/common-adapters/text.d.ts

```diff
@@ -11,8 +12,35 @@ type Background =
   | 'Normal'
   | 'Success'
   | 'Terminal'
+
 type TextType = keyof typeof allTextTypes
 
+export type AllowedColors = Values<
+  Pick<
+    typeof colors,
+    | 'blueDark'
+    | 'blueLighter' //for terminal background only
+    | 'greenDark'
+    | 'redDark'
+    | 'purpleDark'
+    | 'black'
+    | 'black_on_white'
+    | 'black_50'
+    | 'black_50_on_white'
+    | 'black_35'
+    | 'black_20'
+    | 'black_20_on_white'
+    | 'white'
+    | 'white_75'
+    | 'white_40'
+    | 'brown_75'
+    | 'orange'
+    | 'transparent'
+  >
+>
+
+export type StylesTextCrossPlatform = StylesCrossPlatform & {color?: AllowedColors}
```

Comment:
@MarcoPolo particularly this was added because in places this type is passed through props; not sure if that's the best way to do it.

Reply:
Yeah I think that's fine.